torch/testing/_internal/common_methods_invocations.py

```diff
@@ -6246,12 +6246,11 @@ def sample_inputs_sum_to_size(op_info, device, dtype, requires_grad, **kwargs):
     ]
 
     for input_shape, output_shape in sample_shapes:
-        input_t = make_arg(input_shape)
-        yield SampleInput(input_t, args=(output_shape,))
+        yield SampleInput(make_arg(input_shape), args=(output_shape,))
         if output_shape == ():
             continue
-        yield SampleInput(input_t, args=(list(output_shape),))
-        yield SampleInput(input_t, args=(*output_shape,))
+        yield SampleInput(make_arg(input_shape), args=(list(output_shape),))
+        yield SampleInput(make_arg(input_shape), args=(*output_shape,))
```

Comment:
Don't we have a test to catch this one? I thought that the `contiguous` test should have caught this one. @mruberry

Reply:
You mean 

https://github.com/pytorch/pytorch/blob/4bfb7341819b3bfcaf65ddc136f25d23983740a7/test/test_ops.py#L443

?

Reply:
So, it's not clear to Mike or me why the following assert
https://github.com/pytorch/pytorch/blob/4bfb7341819b3bfcaf65ddc136f25d23983740a7/test/test_ops.py#L461-L462
didn't fire in this case.
That assert detects (in a somewhat indirect way) whether there's two sample inputs that share the same tensor. Could you have a look @kshitij12345?

Reply:
We don't really call backward on the output (instead we use `torch.autograd.grad`) and hence we never set `.grad` on the input.

https://github.com/pytorch/pytorch/blob/409efebab8718a7bfc714ab3787e5a8689289697/test/test_ops.py#L519-L526

```python
import torch

t = torch.randn(1, requires_grad=True)
out = torch.sum(t)
grads = torch.autograd.grad(out, t)

print(t.grad) # None
out.backward()
print(t.grad) # tensor([1.])
```

Reply:
oof. I was the one that wrote that bit of code. I'll send a fix if I find the time tomorrow :)